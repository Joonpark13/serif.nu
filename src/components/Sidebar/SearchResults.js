import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import CircularProgress from '@material-ui/core/CircularProgress';

export const styles = {
  loadingContainer: {
    margin: 15,
    display: 'flex',
    justifyContent: 'center',
  },
  searchResults: {
    height: 'calc(100vh - 190px)', // 64px is height of TopBar, 48px is height of SearchBox, 48px is height of tab, 30px margin
    overflow: 'auto',
  },
};

function SearchResults({ searchResults, isFetching, classes }) {
  if (isFetching) {
    return (
      <div className={classes.loadingContainer}>
        <CircularProgress />
      </div>
    );
  }
  return (
    <div className={classes.searchResults}>
      <List>
        {searchResults
          && searchResults.map(course => (
            <ListItem
              key={`${course.school} ${course.subject} ${course.abbv}`}
              button
            >
              <ListItemText primary={course.name} />
            </ListItem>
          ))
        }
      </List>
    </div>
  );
}

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  isFetching: PropTypes.bool.isRequired,
};

export { SearchResults as UnstyledSearchResults };
export default withStyles(styles)(SearchResults);
