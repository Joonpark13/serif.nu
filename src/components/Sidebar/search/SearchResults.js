import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import CircularProgress from '@material-ui/core/CircularProgress';
import Typography from '@material-ui/core/Typography';

export const styles = {
  loadingContainer: {
    margin: 15,
    display: 'flex',
    justifyContent: 'center',
  },
  inputHint: {
    display: 'flex',
    justifyContent: 'center',
  },
};

function SearchResults({
  searchResults,
  isFetching,
  currentSearchInput,
  handleCourseClick,
  classes,
}) {
  if (isFetching) {
    return (
      <div className={classes.loadingContainer}>
        <CircularProgress />
      </div>
    );
  }
  if (searchResults.length === 0 && currentSearchInput.length >= 3) {
    return (
      <Typography className={classes.inputHint}>
        No results
      </Typography>
    );
  }
  return (
    <div>
      {currentSearchInput.length > 0 && currentSearchInput.length <= 2
        && <Typography className={classes.inputHint}> Keep typing!</Typography>
      }
      <List>
        {searchResults
          && searchResults.map(course => (
            <ListItem
              key={`${course.schoolId} ${course.subjectId} ${course.id}`}
              button
              onClick={() => handleCourseClick(course.schoolId, course.subjectId, course.id)}
            >
              <ListItemText primary={`${course.subjectId} ${course.id} ${course.name}`} />
            </ListItem>
          ))
        }
      </List>
    </div>
  );
}

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  isFetching: PropTypes.bool.isRequired,
  handleCourseClick: PropTypes.func.isRequired,
  currentSearchInput: PropTypes.string.isRequired,
};

export { SearchResults as UnstyledSearchResults };
export default withStyles(styles)(SearchResults);
