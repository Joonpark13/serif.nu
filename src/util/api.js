/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
