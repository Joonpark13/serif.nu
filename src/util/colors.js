export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
