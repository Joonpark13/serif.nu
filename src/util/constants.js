/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
