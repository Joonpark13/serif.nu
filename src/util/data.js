import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
