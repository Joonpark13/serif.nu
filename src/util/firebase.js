import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
