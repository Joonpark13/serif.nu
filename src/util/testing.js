/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
