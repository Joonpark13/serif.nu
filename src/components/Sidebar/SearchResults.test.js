import ListItem from '@material-ui/core/ListItem';
import { mockStyles, wrapperCreator } from 'util/testing';
import { UnstyledSearchResults, styles } from './SearchResults';

describe('SearchResults', () => {
  const testResults = [{
    abbv: '629',
    name: 'Employment Law',
    school: 'LAW',
    score: 0.75,
    subject: 'BUSCOM',
    term: '4720',
    type: 'course',
  }, {
    abbv: '631',
    name: 'Entrepreneurship Law',
    school: 'LAW',
    score: 0.75,
    subject: 'BUSCOM',
    term: '4720',
    type: 'course',
  }];
  const classes = mockStyles(styles);

  const defaultProps = {
    searchResults: testResults,
    isFetching: false,
    classes,
    handleCourseClick: () => {},
  };
  const getComponent = wrapperCreator(UnstyledSearchResults, defaultProps);

  it('renders correctly', () => {
    const wrapper = getComponent();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading correctly', () => {
    const wrapper = getComponent({ isFetching: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('handleCourseClick gets called correctly', () => {
    const handleCourseClickMock = jest.fn();
    const wrapper = getComponent({ handleCourseClick: handleCourseClickMock });

    wrapper.find(ListItem).first().simulate('click');

    expect(handleCourseClickMock)
      .toHaveBeenCalledWith(testResults[0].school, testResults[0].subject, testResults[0].abbv);
  });
});
