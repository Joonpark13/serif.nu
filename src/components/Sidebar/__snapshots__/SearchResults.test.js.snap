// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    classes={
      Object {
        "loadingContainer": "",
        "searchResults": "",
      }
    }
    handleCourseClick={[Function]}
    isFetching={false}
    searchResults={
      Array [
        Object {
          "abbv": "629",
          "name": "Employment Law",
          "school": "LAW",
          "score": 0.75,
          "subject": "BUSCOM",
          "term": "4720",
          "type": "course",
        },
        Object {
          "abbv": "631",
          "name": "Entrepreneurship Law",
          "school": "LAW",
          "score": 0.75,
          "subject": "BUSCOM",
          "term": "4720",
          "type": "course",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(List)>
        <WithStyles(ListItem)
          button={true}
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            primary="BUSCOM 629 Employment Law"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          button={true}
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            primary="BUSCOM 631 Entrepreneurship Law"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItem)
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ListItemText)
              primary="BUSCOM 629 Employment Law"
            />
          </WithStyles(ListItem)>,
          <WithStyles(ListItem)
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ListItemText)
              primary="BUSCOM 631 Entrepreneurship Law"
            />
          </WithStyles(ListItem)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "LAW BUSCOM 629",
          "nodeType": "class",
          "props": Object {
            "button": true,
            "children": <WithStyles(ListItemText)
              primary="BUSCOM 629 Employment Law"
            />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "primary": "BUSCOM 629 Employment Law",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "LAW BUSCOM 631",
          "nodeType": "class",
          "props": Object {
            "button": true,
            "children": <WithStyles(ListItemText)
              primary="BUSCOM 631 Entrepreneurship Law"
            />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "primary": "BUSCOM 631 Entrepreneurship Law",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(List)>
          <WithStyles(ListItem)
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ListItemText)
              primary="BUSCOM 629 Employment Law"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ListItemText)
              primary="BUSCOM 631 Entrepreneurship Law"
            />
          </WithStyles(ListItem)>
        </WithStyles(List)>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItem)
              button={true}
              onClick={[Function]}
            >
              <WithStyles(ListItemText)
                primary="BUSCOM 629 Employment Law"
              />
            </WithStyles(ListItem)>,
            <WithStyles(ListItem)
              button={true}
              onClick={[Function]}
            >
              <WithStyles(ListItemText)
                primary="BUSCOM 631 Entrepreneurship Law"
              />
            </WithStyles(ListItem)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "LAW BUSCOM 629",
            "nodeType": "class",
            "props": Object {
              "button": true,
              "children": <WithStyles(ListItemText)
                primary="BUSCOM 629 Employment Law"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "BUSCOM 629 Employment Law",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "LAW BUSCOM 631",
            "nodeType": "class",
            "props": Object {
              "button": true,
              "children": <WithStyles(ListItemText)
                primary="BUSCOM 631 Entrepreneurship Law"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "BUSCOM 631 Entrepreneurship Law",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SearchResults renders loading correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    classes={
      Object {
        "loadingContainer": "",
        "searchResults": "",
      }
    }
    handleCourseClick={[Function]}
    isFetching={true}
    searchResults={
      Array [
        Object {
          "abbv": "629",
          "name": "Employment Law",
          "school": "LAW",
          "score": 0.75,
          "subject": "BUSCOM",
          "term": "4720",
          "type": "course",
        },
        Object {
          "abbv": "631",
          "name": "Entrepreneurship Law",
          "school": "LAW",
          "score": 0.75,
          "subject": "BUSCOM",
          "term": "4720",
          "type": "course",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(CircularProgress) />,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(CircularProgress) />,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
